src/test/java/org/dcsa/api_validator/bkg/v1/cancellations/CancelBookingTest.java

```diff
@@ -0,0 +1,164 @@
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static io.restassured.RestAssured.given;
+import static org.dcsa.api_validator.bkg.v1.BookingTestConfiguration.*;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+
+public class CancelBookingTest {
+
+	JSONObject cancellationRequest;
+	String bookingAllowedToCancel;
+	List<String> bookingsNotAllowedToCancel;
+
+	@BeforeClass
+	private void createCancellationRequest() throws JsonProcessingException {
+		List<String> allowedStatusses = Arrays.asList("RECE", "PENU", "CONF");
+		JSONObject cancellation = new JSONObject();
+		cancellation.put("documentStatus", "CANC");
+		cancellation.put("reason", "booking cancelled during test");
+		cancellationRequest = cancellation;
+
+		bookingAllowedToCancel = createBooking().get("carrierBookingRequestReference").toString();
+		bookingsNotAllowedToCancel = findAllBookings().stream().filter(map -> !allowedStatusses.contains(map.get("documentStatus"))).map(map -> String.valueOf(map.get("carrierBookingRequestReference"))).collect(Collectors.toList());
+	}
+
+	//Create a minimal booking to cancel, so it does not effect the base test data
+	private Map createBooking() throws JsonProcessingException {
+
+		Map<String, Object> bookingRequest = new LinkedHashMap<>();
+		bookingRequest.put("receiptTypeAtOrigin", "CY");
+		bookingRequest.put("deliveryTypeAtDestination", "CY");
+		bookingRequest.put("cargoMovementTypeAtOrigin", "FCL");
+		bookingRequest.put("cargoMovementTypeAtDestination", "LCL");
+		bookingRequest.put("serviceContractReference", "Test");
+		bookingRequest.put("isPartialLoadAllowed", true);
+		bookingRequest.put("isExportDeclarationRequired", false);
+		bookingRequest.put("isImportLicenseRequired", false);
+		bookingRequest.put("submissionDateTime", "2021-11-03T11:41:00+01:00");
+		bookingRequest.put("communicationChannel", "AO");
+		bookingRequest.put("expectedDepartureDate", "2021-05-17");
+		bookingRequest.put("isEquipmentSubstitutionAllowed", true);
+
+		Map<String, Object> commodity = new LinkedHashMap<>();
+		commodity.put("commodityType", "Bloom");
+		commodity.put("cargoGrossWeight", 2000.0);
+		commodity.put("cargoGrossWeightUnit", "LBR");
+    bookingRequest.put("commodities", Arrays.asList(commodity));
+
+		ObjectMapper objectMapper = new ObjectMapper();
+		String jsonBookingRequest = objectMapper.writeValueAsString(bookingRequest);
+
+		return given()
+			.auth()
+			.oauth2(Configuration.accessToken)
+			.contentType(ContentType.JSON)
+			.body(jsonBookingRequest)
+			.post(Configuration.ROOT_URI + BOOKING_PATH)
+			.then()
+			.assertThat()
+			.statusCode(202)
+			.body(JSON_SCHEMA_VALIDATOR)
+			.extract().body().as(Map.class);
+	}
+
+	private List<Map> findAllBookings() {
+		return given()
+			.auth()
+			.oauth2(Configuration.accessToken)
+			.get(Configuration.ROOT_URI + BOOKING_SUMMARIES_PATH)
+			.then()
+			.assertThat()
+			.statusCode(200)
+			.body("size()", greaterThanOrEqualTo(0))
```

Comment:
Should this be `greaterThanOrEqualTo(1)` given that `testCancelBookingInInvalidDocumentStatus` will fail with `IndexOutOfBoundsException` if there are no bookings?

Reply:
Hah, turns out the CI fails for exactly this reason. :D